### PR TITLE
40-0247 - Add Flipper functionality

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Add Flipper functionality to PMC (40-0247)
- Team: Veteran Facing Forms
- This is the Flipper implementation; end-date TBD

## Related PR

[vets-api](https://github.com/department-of-veterans-affairs/vets-api/pull/14409)
[content-build](https://github.com/department-of-veterans-affairs/content-build/pull/1777)

## Related issue(s)

- _Link to ticket created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#484

## Testing done

- Local browser-, unit-, & e2e-testing were all successful

## Screenshots## What areas of the site does it impact?

This form ONLY

### CODE REVIEW

**Comment by tlei123** [comment]
@Thrillberg , I'm hoping you wouldn't mind testing this along with vets-api, since I've never been able to start that b/e app locally.  Both feature-branches have the same name:

this feature-branch: va-forms/20-0247/484-flipper
vets-api feature-branch: va-forms/20-0247/484-flipper

Thanks much in advance!

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            href="/find-forms/about-form-40-0247"
+            text="Get VA Form 40-0247 to download"
+          />
+        </p>
+        <p>
+          Then, upload your completed application and supporting documents on
+          the AccessVA website using the QuickSubmit tool.
+        </p>
+        <p>
+          If it’s your first time signing into this tool, you’ll need to
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:72]: Could we add a space between in and to? It should be "...signing in to this..." @tlei123

Comment (megzehn) [reply src/applications/static-pages/simple-forms/40-0247/App.js:72]: I sent it the wrong way in the initial copy. Sorry about that!

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            href="/find-forms/about-form-40-0247"
+            text="Get VA Form 40-0247 to download"
+          />
+        </p>
+        <p>
+          Mail your application, along with copies of the Veteran’s death
+          certificate and DD214 or other discharge documents, to this address:
+        </p>
+        <p>
+          NCA FP Evidence Intake Center
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:40]: is it possible to use the address component here? @tlei123 @tbaker1026 

https://design.va.gov/components/address-block

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            text="Find your nearest VA regional office"
+          />
+        </p>
+        <h3>Option 5: Fax your application</h3>
+      </>
+    );
+  }
+
+  return (
+    <>
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:96]: Could we add this line before "Option 1"? @tlei123 @tbaker1026 
<p>You can apply for a PMC in any of these 4 ways.</p>

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+          href="/find-forms/about-form-40-0247"
+          text="Get VA Form 40-0247 to download"
+        />
+      </p>
+      <p>
+        Then, upload your completed application and supporting documents on the
+        AccessVA website using the QuickSubmit tool.
+      </p>
+      <p>
+        If it’s your first time signing into this tool, you’ll need to register
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:113]: Could we add a space between in and to? It should be "...signing in to this..." @tlei123

My fault for missing this in the copy!

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          href="/find-forms/about-form-40-0247"
+          text="Get VA Form 40-0247 to download"
+        />
+      </p>
+      <p>
+        Mail your application, along with copies of the Veteran’s death
+        certificate and DD214 or other discharge documents, to this address:
+      </p>
+      <p>
+        NCA FP Evidence Intake Center
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:139]: Same question as above. Is it possible to use the address component here for these 3 lines? @tlei123 @tbaker1026

https://design.va.gov/components/address-block

**Review by megzehn — COMMENTED** [review]
Thanks @tlei123 and @tbaker1026 for meeting with me! I reviewed the content and found a few mistakes that I passed on to you in my original doc--sorry about that! I left corrections in the comments. Let me know if you want me to commit them.

We also have an address in both parts of this widget. Is it a big lift to use the address component for that?

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    return <va-loading-indicator message="Loading..." />;
+  }
+
+  if (formEnabled) {
+    return (
+      <>
+        <p>You can apply for a PMC in any of these 5 ways.</p>
+        <h3>Option 1: Apply online</h3>
+        <a
+          className="vads-c-action-link--blue"
```
Comment (megzehn) [inline src/applications/static-pages/simple-forms/40-0247/App.js:19]: Could you make this link green? It should be a primary action link. https://design.va.gov/components/link/action#primary

@tlei123

**Review by megzehn — COMMENTED** [review]
Added one more comment!

File: src/applications/static-pages/simple-forms/40-0247/App.js (hunk 1 above)
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:72]: @megzehn @tbaker1026 , yes, and done.

File: src/applications/static-pages/simple-forms/40-0247/App.js (hunk 3 above)
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:96]: @megzehn , yes; added.

File: src/applications/static-pages/simple-forms/40-0247/App.js (hunk 2 above)
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:40]: @megzehn , yes; className="va-address-block" added to the <p> tag.  It'll looks like an [address block](https://design.va.gov/storybook/?path=/docs/components-address-block--default-story) on your page.

File: src/applications/static-pages/simple-forms/40-0247/App.js (hunk 5 above)
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:139]: @megzehn , yes; className="va-address-block" added to the <p> tag.  It'll looks like an [address block](https://design.va.gov/storybook/?path=/docs/components-address-block--default-story) on your page.

File: src/applications/static-pages/simple-forms/40-0247/App.js (hunk 6 above)
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:19]: @megzehn , yes; className changed to "vads-c-action-link--grean".

**Comment by tlei123** [comment]
@megzehn @tbaker1026 , all request changes are now done.  Feel free to check the [file-diff and resolved-conversations](https://github.com/department-of-veterans-affairs/vets-website/pull/26571/files#diff-68447ed9fa33aeb1c155c4afe282e894d1cc19ec65d1cd53831c327eabc8e11b) again.

**Review by Thrillberg — COMMENTED** [review]
Overall looks good to me but I've got a few questions inside!

File: src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js [hunk 7]
```diff
@@ -115,6 +116,18 @@ const testConfig = createTestConfig(
           });
         });
       },
+      [pagePaths.certificatesPage]: ({ afterHook }) => {
+        afterHook(() => {
+          cy.get('@testData').then(data => {
+            const { certificates } = data;
+
+            fillTextWebComponent('certificates', certificates);
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js:124]: reminder that if you use this new property, this page and field should be detected and tested automatically (runs on any page you haven't manually specified)
`useWebComponentFields: true`

**Review by rhasselle-oddball — COMMENTED** [review]
changes look good to me

File: src/applications/simple-forms/21-0845/utils.js [hunk 8]
```diff
@@ -55,3 +55,19 @@ export const getEnumsFromConstants = (constants = {}) => {
 
   return enums;
 };
+
+export const objHasEmptyValues = obj => {
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/21-0845/utils.js:59]: Removed.  Thanks for catching this.  Not sure why this snuck in here, but you're right it's no longer used.

File: src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js (hunk 7 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js:124]: Prop added.  Thanks!

File: src/applications/static-pages/simple-forms/40-0247/App.js [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+
+const App = ({ formEnabled }) => {
+  if (formEnabled === undefined) {
+    return <va-loading-indicator message="Loading..." />;
+  }
+
+  if (formEnabled) {
+    return (
+      <>
+        <p>You can apply for a PMC in any of these 5 ways.</p>
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/static-pages/simple-forms/40-0247/App.js:16]: This needs to be set up by Content team after merge.  It goes into their content-page.

File: src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js [hunk 10]
```diff
@@ -143,6 +156,7 @@ const testConfig = createTestConfig(
     },
 
     setupPerTest: () => {
+      cy.intercept('GET', '/v0/feature_toggles?*', featureToggles);
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/40-0247/tests/e2e/0247-pmc.cypress.spec.js:159]: No, 'cuz in CI the test runs just with vets-website, w/o vets-api.  We need this mock in the test as long as we're using Flipper in the form.

**Review by Thrillberg — APPROVED** [review]

**Review by powellkerry — APPROVED** [review]

**Pull request merged**